From cc3758e0174efad8220fe5f68ded750a39a9491b Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 24 May 2018 13:44:30 +0300
Subject: [PATCH 1475/1921] dts: mvebu: Update A8K AP806/AP807 SDHCI settings

Select the AP SDHCI PHY slow mode for AP806 die only (move it
from armada-ap80x.dtsi to armada-ap806.dtsi). This will allow
running AP807 based devices at HS400 speed.
Remove Ap SDHCI slow mode property from MacchiatoBin board DTS
since it is already selected on the SoC level.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I4e8baa2bbfb1afea55d7aef22090196732a4ef9c
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/50234
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dtsi |  5 -----
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 12 ++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi      |  1 -
 3 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dtsi
index 4d26e99f8db5..2bc1e03542c7 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dtsi
@@ -106,11 +106,6 @@
 
 &ap_sdhci0 {
 	bus-width = <8>;
-	/*
-	 * Not stable in HS modes - phy needs "more calibration", so add
-	 * the "slow-mode" and disable SDR104, SDR50 and DDR50 modes.
-	 */
-	marvell,xenon-phy-slow-mode;
 	no-1-8-v;
 	no-sd;
 	no-sdio;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 866628679ac7..828cd539173b 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -28,3 +28,15 @@
 		reg = <0x278 0xa30>;
 	};
 };
+
+&ap_sdhci0 {
+	/*
+	 * SoC based on AP806 revision A0, A1 and A2 should use slow mode
+	 * settings for Ap SDHCI due to HW Erratum HWE-7296210
+	 * AP806 revesion B0 and later has this erratum fixed and the slow
+	 * mode could be removed in board DTS:
+	 *     /delete-property/marvell,xenon-phy-slow-mode;
+	 * Starting from B0 revision, the AP SDHCI can run with HS400 timing.
+	 */
+	marvell,xenon-phy-slow-mode;
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index 6343a9b54289..e62a00858f4b 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -300,7 +300,6 @@
 				clocks = <&ap_clk 4>;
 				dma-coherent;
 				iommus = <&smmu 0x5>;
-				marvell,xenon-phy-slow-mode;
 				status = "disabled";
 			};
 
-- 
2.31.1

